pkg/kubelet/server.go

```diff
@@ -260,12 +260,12 @@ func (s *Server) handleContainerLogs(w http.ResponseWriter, req *http.Request) {
 
 // handleBoundPods returns a list of pod bound to the Kubelet and their spec
 func (s *Server) handleBoundPods(w http.ResponseWriter, req *http.Request) {
-	pods, err := s.host.GetBoundPods()
+	pods, err := s.host.GetPods()
 	if err != nil {
 		s.error(w, err)
 		return
 	}
-	boundPods := &api.BoundPods{
```

Comment:
@erictune's worry is about external clients. AFAIK the Kubelet API is not meant for public consumption so this change is probably okay. To be extra safe we can change the endpoint name /boundPods -> /pods.